src/python/grpcio/grpc/_cython/_cygrpc/records.pyx.pxi

```diff
@@ -235,18 +235,13 @@ cdef class ByteBuffer:
     if data is None:
       self.c_byte_buffer = NULL
       return
-    if isinstance(data, bytes):
-      pass
-    elif isinstance(data, basestring):
-      data = data.encode()
-    elif isinstance(data, ByteBuffer):
+    if isinstance(data, ByteBuffer):
```

Comment:
This is the only function that should accept a ByteBuffer AFAICT.  I'm fine with having the one-off handling built right into the method.